lib/src/component/abstract_transition.dart

```diff
@@ -368,6 +369,23 @@ abstract class AbstractTransitionComponent<T extends AbstractTransitionProps,
     }
   }
 
+  /// Returns attributes only available during testing that indicate the state of the transition.
+  Map<String, String> getTransitionTestAttributes() {
+    if (!component_base.UiProps.testMode) return const {};
+
+    const enumToAttrValue = const <TransitionPhase, String>{
+      TransitionPhase.SHOWN: 'shown',
+      TransitionPhase.HIDDEN: 'hidden',
+      TransitionPhase.HIDING: 'hiding',
+      TransitionPhase.PRE_SHOWING: 'pre-showing',
+      TransitionPhase.SHOWING: 'showing',
+    };
+
+    return {
+      'data-transition-phase': enumToAttrValue[state.transitionPhase],
```

Comment:
@greglittlefield-wf should we store `data-transition-phase` as a constant sorta like we do for the test id attribute?